### ISSUE TITLE
Symbol annotations are clipped to their corresponding tile

### ISSUE THREAD

**Issue by brunoabinader** [mapbox/mapbox-gl-native#5419]
Example code:
```
test.map.setStyleJSON(util::read_file("test/fixtures/api/empty.json"));
test.map.addAnnotationIcon("default_marker", namedMarker("default_marker.png"));
test.map.addAnnotation(SymbolAnnotation { Point<double>(0, 0), "default_marker" });

test.map.setZoom(test.map.getMaxZoom());
test.checkRendering("point_annotation_max_zoom");
```
| Expected result | Actual result |
| ---- | ---- |
| ![expected](https://cloud.githubusercontent.com/assets/76133/16223789/bda1776c-37a7-11e6-82bd-e5e554086473.png) | ![actual](https://cloud.githubusercontent.com/assets/76133/16223802/d14d77f2-37a7-11e6-896c-2f180fa73391.png) |

/cc @mapbox/gl

**Comment by brunoabinader** [comment]
Related: https://github.com/mapbox/mapbox-gl-native/issues/1673

**Comment by brunoabinader** [comment]
This is fixed in #3563.
